scisample/uqpipeline_sampler.py

```diff
@@ -0,0 +1,172 @@
+"""
+Module defining the UQ Pipeline sampler object.
+"""
+
+import logging
+import random
+import os
+import sys
+from contextlib import suppress
+
+UQPIPELINE_SAMPLE = False
+UQPIPELINE_SAMPLE_PATH = '/collab/usr/gapps/uq/UQPipeline/smplg_cmpnt'
+if os.path.exists(UQPIPELINE_SAMPLE_PATH):
+    sys.path.append(UQPIPELINE_SAMPLE_PATH)
+    with suppress(ModuleNotFoundError):
+        import sampling.sampler as sampler
+        import sampling.composite_samples as composite_samples
+        UQPIPELINE_SAMPLE = True
+
+from scisample.base_sampler import BaseSampler
+from scisample.utils import log_and_raise_exception
+from scisample.utils import test_for_uniform_lengths, test_for_min_max
+
+# @TODO: can this duplicate code be removed?
+
+LOG = logging.getLogger(__name__)
+
+class UQPipelineSampler(BaseSampler):
+    """
+    Class which wraps UQPipeline sampling methods.
+
+    This class currently supports two ways of creating samples with 
+    UQPipeline methods:
+
+    1. With "sampler.<SAMPLER_NAME>.sample_points" methods, and
+    2. with the "composite_samples" class.
+
+    .. code:: yaml
+
+         # `sample_points` method example.
+         sampler:
+            type: uqpipeline
+            uq_points: points
+            uq_variables: ['X1', 'type']
+            uq_code: |

```

Comment:
It's generally discouraged to use `eval` to evaluate arbitrary code.  If this is a consistent structure, where `sampler.SamplerType.sample_points(**kwargs)` will be called, then I would recommend spelling it out:
```
    sampler_type: CartesianCrossProduct
   sample_points_kwargs:
        num_divisions: [3, 3]
        box:
          - [-1, 1]
          - []
         values:
           - []
           - ['foo', 'bar']
```
Then instead of using eval, use
```
uq_sampler = getattr(sampler, self.data['sampler_type'])
points = uq_sampler.sample_points(**self.data['sample_points_kwargs']
```
@jsemler or @FrankD412 do you have alternate suggestions?